src/parser/parse_tree_cl.c

```diff
@@ -14080,10 +14086,16 @@ pt_print_select (PARSER_CONTEXT * parser, PT_NODE * p)
 	  if (p->info.query.q.select.hint & PT_HINT_ORDERED)
 	    {
 	      /* force join left-to-right */
-	      q = pt_append_nulstring (parser, q, "ORDERED");
-	      if (p->info.query.q.select.ordered)
+	      q = pt_append_nulstring (parser, q, "ORDERED ");
+	    }
+
+	  if (p->info.query.q.select.hint & PT_HINT_LEADING)
+	    {
+	      /* force join left-to-right */
+	      q = pt_append_nulstring (parser, q, "LEADING");
```

Comment:
If `p->info.query.q.select.leading` is NULL, I think it is better not to output `/*+ LEADING */` either.

Reply:
Even if it doesn't work, I think it's right to show the content as written by the user.